recipes/dhcpcd/files/init

```diff
@@ -16,7 +16,12 @@ do_start() {
 }
 
 do_stop() {
-    start-stop-daemon -K --pidfile $PIDFILE --exec $DAEMON
+    # only try to stop the daemon if there is actually something
+    # running. Otherwise a restart before a start will fail
+    start-stop-daemon -K -o --pidfile $PIDFILE --exec $DAEMON
+    if [ "$?" != "0" ] ; then
+	start-stop-daemon -K --pidfile $PIDFILE --exec $DAEMON
+    fi
 }
```

Comment:
Does this work? You still have `set -e`, so if the first start-stop-daemon fails, you should never get around to testing $?. At least that how a quick test in bash behaves. If you make the do_stop part of a conditional chain, the set -e behaviour is suppressed, so it seems (again, tested in bash) that one can do "do_stop || true" in the restart) case. But then you don't need to change do_stop itself.